src/builtins.c

```diff
@@ -1655,6 +1655,46 @@ JL_CALLABLE(jl_f__equiv_typedef)
     return equiv_type(args[0], args[1]) ? jl_true : jl_false;
 }
 
+JL_CALLABLE(jl_f_get_binding_type)
+{
+    JL_NARGS(get_binding_type, 2, 2);
+    JL_TYPECHK(get_binding_type, module, args[0]);
+    JL_TYPECHK(get_binding_type, symbol, args[1]);
+    jl_value_t* ty = jl_binding_type((jl_module_t*)args[0], (jl_sym_t*)args[1]);
+    return ty;
+}
+
+JL_CALLABLE(jl_f_set_binding_type)
+{
+    JL_NARGS(set_binding_type!, 3, 3);
+    JL_TYPECHK(set_binding_type!, module, args[0]);
+    JL_TYPECHK(set_binding_type!, symbol, args[1]);
+    jl_value_t *ty = args[2];
+    JL_TYPECHK(set_binding_type!, type, ty);
+    jl_binding_t *b = jl_get_binding_wr((jl_module_t*)args[0], (jl_sym_t*)args[1], 1);
+    if (b->constp && ty != (jl_value_t*)jl_any_type) {
```

Comment:
This check seems unnecessary. Sure, declaring a type on a constant is kind of redundant but we might as well allow it.

Reply:
Yeah, I think we can remove that check. Should declaring typed globals as const also be allowed then?